src/porepy/fracs/fractures_2d.py

```diff
@@ -171,18 +200,282 @@ def mesh(self, mesh_args, tol=None, do_snap=True, constraints=None, **kwargs):
             tol = self.tol
         if constraints is None:
             constraints = np.empty(0, dtype=np.int)
+        else:
+            constraints = np.atleast_1d(constraints)
+        constraints = np.sort(constraints)
+
+        if file_name is None:
+            file_name = "gmsh_frac_file"
+        in_file = file_name + ".geo"
+        out_file = file_name + ".msh"
 
         p = self.pts
         e = self.edges
 
+        # Snap points to edges
         if do_snap and p is not None and p.size > 0:
-            p, _ = pp.frac_utils.snap_fracture_set_2d(p, e, snap_tol=tol)
-        grid_list = porepy.fracs.simplex.triangle_grid(
-            p, e[:2], self.domain, tol=tol, constraints=constraints, **mesh_args
+            p, _ = pp.frac_utils.snap_fracture_set_2d(p, self.edges, snap_tol=tol)
+
+        self.pts = p
+
+        if not self.bounding_box_imposed:
+            edges_deleted = self.impose_external_boundary(self.domain)
+
+            # Find edges of constraints to delete
+            to_delete = np.where(np.isin(constraints, edges_deleted))[0]
+
+            # Adjust constraint indices for deleted edges
+            adjustment = np.zeros(constraints.size, dtype=np.int)
+            for e in edges_deleted:
```

Comment:
do we want to keep a numeration of the branches that stays the same even if some edges are deleted?